frontend/src/metabase/query_builder/components/view/View.jsx

```diff
@@ -422,13 +431,43 @@ class View extends Component {
     const { isNative } = Lib.queryDisplayInfo(question.query());
 
     const isNewQuestion = !isNative && Lib.sourceTableOrCardId(query) === null;
-    const isModelOrMetric =
-      question.type() === "model" || question.type() === "metric";
+    const isModel = question.type() === "model";
+    const isMetric = question.type() === "metric";
 
-    if (isModelOrMetric && queryBuilderMode === "dataset") {
+    if ((isModel || isMetric) && queryBuilderMode === "dataset") {
       return (
         <>
-          <DatasetEditor {...this.props} />
+          {isModel && <DatasetEditor {...this.props} />}
+          {isMetric && (
+            <MetricEditor
```

Comment:
The idea is to have all QB-specific logic in the QB itself. E.g. the QB should know that it needs to switch to the "view" mode when the metric has been successfully saved.